server/protocols/webadmin-cli/README.md

```diff
@@ -123,4 +151,171 @@ Show all users' name on the list.
+
+## Manage Mailboxes
+
+### Create a mailbox
+Create a specific mailbox for a user:
+
+```
+{cli} mailbox create <username> <mailboxName>
+```
+
+Resource name username should be an existing user.
+
+Resource name mailboxName should not be empty, nor contain # & % * characters.
+
+To create nested mailboxes, for instance a work mailbox inside the INBOX mailbox, people should use the . separator. E.g: INBOX.work
+
+### Test a mailbox existence
+
+```
+{cli} mailbox exist <usernameToBeUsed> <mailboxNameToBeTested>
+```
+
+Resource name usernameToBeUsed should be an existing user 
+
+Resource name mailboxNameToBeTested should not be empty
+
+### Delete a mailbox and its children
+
+```
+{cli} mailbox delete <usernameToBeUsed> --mailbox <mailboxNameToBeDeleted>
+```
+
+Resource name usernameToBeUsed should be an existing user 
+
+Resource name mailboxNameToBeDeleted should not be empty
+
+### Delete all mailboxes of a user
+
+```
+{cli} mailbox delete <usernameToBeUsed>
+```
+
+Resource name usernameToBeUsed should be an existing user
+
+### Get mailboxes list
+
+```
+{cli} mailbox list <usernameToBeUsed>
+```
+
+Resource name usernameToBeUsed should be an existing user
+
+### Reindex a user mailboxes
+
+```
+{cli} mailbox reindex --username <usernameToBeUsed>
+```
+
+Will schedule a task for reIndexing all the mails in “usernameToBeUsed” mailboxes.
+
+Resource name usernameToBeUsed should be an existing user
+
+### Reindex all users mailboxes
+
+```
+{cli} mailbox reindex
+```
+
+Will schedule a task for reIndexing all the mails stored on this James server.
+
+## Manage Domain Mappings
+
+Given a configured source (from) domain and a destination (to) domain, when an email is sent to an address belonging to the source domain, then the domain part of this address is overwritten, the destination domain is then used. A source (from) domain can have many destination (to) domains.
+
+For example: with a source domain james.apache.org maps to two destination domains james.org and apache-james.org, when a mail is sent to admin@james.apache.org, then it will be routed to admin@james.org and admin@apache-james.org
+
+This feature uses [Recipients rewrite table](https://james.apache.org/server/config-recipientrewritetable.html) and requires the [RecipientRewriteTable](https://github.com/apache/james-project/blob/master/server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/RecipientRewriteTable.java) mailet to be configured.
+
+Note that email addresses are restricted to ASCII character set. Mail addresses not matching this criteria will be rejected.
+
+### Listing all domain mappings
+
+Show all configured domain mappings
+```
+{cli} domain-mapping list
+```
+
+### Listing all destination domains for a source domain
+
+With sourceDomain.tld as the value passed to fromDomain resource name, this command will show all destination domains configured to that domain
+```
+{cli} domain-mapping list --sourcedomain <sourceDomain.tld>
```

Comment:
Why --sourcedomain only here ?

Reply:
Same as above, to put "'Show all configured domain mappings" and "Listing all destination domains for a source domain" in 1 command